tests/type_propagation/gold/default-test.gold-O2

```diff
@@ -8,18 +8,12 @@ main tests/type_propagation/default-test.toit
 
 foo tests/type_propagation/default-test.toit
  - argument 0: {False}
-  0[016] - load local 2
-  1[084] - branch if not null T4
-  4[016] - load local 2
-  5[083] - branch if false T8
-  8[092] - return null S0 1
+  0[092] - return null S0 1
 
 foo2 tests/type_propagation/default-test.toit
  - argument 0: {False}
   0[016] - load local 2
   1[084] - branch if not null T8
```

Comment:
We still need to capture the propagated types for identical and branch-if-null, so we can eliminate this branch (we know that false is not null).